### PR TITLE
Tbolt/3257 state admin cert endpoint

### LINKED ISSUE #3257: [Feature] Create endpoint to add authorization record to the State Admin Certification Table
**Description and related issues -**
We need an endpoint for adding certifications to the State Admin Certification Table. 

Database design

The endpoint **should**:
- be restricted to 'edit-state-certification' permission
- endpoint should handle the [latest form design](https://www.figma.com/file/6eVvo7JjvXiovGTR4BoVgK/CMS-eAPD-2021?node-id=5386%3A34533) which includes:
  - State employee to be delegated eAPD State Administrator
  - State
  - State employee email address
  - State employee phone number
  - Name of person who completed the delegation letter
  - Title of the person who completed the delegation letter
  - Email address of the person who completed the delegation letter
  - Signature (printed name on the delegation letter)
  - File URL of the letter
- set the certification date to today and the expiration to a year from today
- update the `state_admin_certifications_audit` table to record the id of the record that was changed in the `state_admin_certification` table

### Acceptance criteria

| Given | When | Then |
| ----- | ---- | ---- |
| a user with  'edit-state-certification' permission  | posting valid data | return 200 |
| a user with  'edit-state-certification' permission | posting invalid data | return 400 |
| user without  'edit-state-certification' permission | posting any kind of data | return 403 |
| not a logged in user | posting any kind of data | return 401 |

### This task is done when…

- [ ] all the requirements are met
- [ ] all acceptance criteria are met

### PR DESCRIPTION
**Pull request by tbolt**
Resolves #3257 

**Description-**
Adds endpoint `/auth/certifications` to allow a Fed Admin to submit a state admin certification letter

**This pull request was tested in the follow ways…**

- Using an HTTP Client, tested with no jwt, a not logged-in jwt, an un-authorized jwt, and a valid jwt
- With a valid token, tested with a sample payload with both valid and invalid data

**Steps to manually verify this change...**
1. Using an HTTP Client first attempt to POST to the endpoint `auth/certifications` with no valid jwt, verify a `401` is thrown
2. Then attempt with a non-Federal admin role, and verify a `403`
3. Use the sample JSON below and verify a `200`
4. Attempt the above again and verify `400` as the email + state must be unique

Sample JSON Body
```
{
  "ffy": "2021"
  "name": "Roger Klotz",
  "email": "fake@email.com",
  "phone": "4105555555",
  "state": "md",
  "fileUrl": "/auth/certifications/files/12345"
}
```
### This pull request is ready to review when...

- [ ] Automated tests are updated (and all tests are passing)
- [ ] The change has been documented
- [ ] Associated OpenAPI documentation has been updated
- [ ] Changelog is updated as appropriate
- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

File: api/seeds/shared/set-up-users.js [hunk 1]
```diff
@@ -89,17 +88,6 @@ const createUsersToAdd = async (knex, oktaClient) => {
       status: 'approved',
       username: 'em@il.com'
     });
-    // Add an expired certification and this user will be downgraded to "regular user"
```
Comment (thetif) [inline api/seeds/shared/set-up-users.js:92]: are you going to seed this data anywhere? or do you want to add it in a future ticket when we need some examples of matched certifications and users.

**Review by thetif — APPROVED** [review]

File: api/package.json [hunk 2]
```diff
@@ -40,6 +40,7 @@
     "Nicolas Aretakis",
     "Akua Amponsah",
     "Heather Battaglia",
+    "Ty Bolt",
```
Comment (knollfear) [inline api/package.json:43]: welcome to the team

File: api/routes/auth/certifications/files.endpoint.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    unauthenticatedTest('post', url);
+    unauthorizedTest('post', url);
+
+    describe('when authenticated as a user with permission', () => {
+      let api;
+      beforeAll(async () => {
+        api = login();
+      });
+
+      it('with a non-existant apd ID', async () => {
```
Comment (knollfear) [inline api/routes/auth/certifications/files.endpoint.js:58]: probably not an APD id.  I also looked it up and existent is the right spelling  https://en.wiktionary.org/wiki/existant

File: api/routes/auth/certifications/files.test.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      );
+    });      
+  });    
+  
+  endpointTest.test('GET endpoint for fetching a state certification letter/file', async tests => {
+    let handler; 
+    
+    tests.beforeEach(async () => {
+      filesEndpoint(app, { ...di });
+      handler = app.get.args[0].pop();
```
Comment (knollfear) [inline api/routes/auth/certifications/files.test.js:131]: This should be a named lookup

File: api/seeds/shared/set-up-users.js (hunk 1 above)
Comment (tbolt) [reply api/seeds/shared/set-up-users.js:92]: So we previously had certifications attached to 2 non-eAPD State Admin users. I removed those as they didn't make sense. I removed the `state_admin_certifications_audit` (and related seed data) based on our dev sync. Finally, I updated the certification that is attached to our eAPD State Admin user. So this should be correct given where we are at in the fed admin feature implementation.

File: api/routes/auth/certifications/files.endpoint.js (hunk 3 above)
Comment (tbolt) [reply api/routes/auth/certifications/files.endpoint.js:58]: good catch. this was copy + pasted over from another test. Not an actual relevant test I wrote or want included so I will take it out.

Comment (tbolt) [reply api/routes/auth/certifications/files.endpoint.js:58]: good call, thats what I get for not thoroughly reading copy + pasted code 🤣

File: api/routes/auth/certifications/files.test.js (hunk 4 above)
Comment (tbolt) [reply api/routes/auth/certifications/files.test.js]: This code was originally created in 3403 and it was updated there before merge so this should be good now.

File: api/routes/auth/certifications/post.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+          name, 
+          email, 
+          phone, 
+          state, 
+          fileUrl,
+          uploadedBy: req.user.id,
+          uploadedOn: new Date()
+        });
+        if (error) {
+          res.status(400).end();
```
Comment (knollfear) [inline api/routes/auth/certifications/post.js:41]: should this not trigger next to kick it over to the error handler?

**Review by knollfear — APPROVED** [review]
LGTM!

**Pull request merged**